samples/api/hpp_dynamic_uniform_buffers/hpp_dynamic_uniform_buffers.cpp

```diff
@@ -0,0 +1,480 @@
+/* Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Demonstrates the use of dynamic uniform buffers, using vulkan.hpp
+ *
+ * Instead of using one uniform buffer per-object, this example allocates one big uniform buffer
+ * with respect to the alignment reported by the device via minUniformBufferOffsetAlignment that
+ * contains all matrices for the objects in the scene.
+ *
+ * The used descriptor type vk::DescriptorType::eUniformBufferDynamic then allows to set a dynamic
+ * offset used to pass data from the single uniform buffer to the connected shader binding point.
+ */
+
+#include <benchmark_mode/benchmark_mode.h>
+#include <hpp_dynamic_uniform_buffers.h>
+
+HPPDynamicUniformBuffers::HPPDynamicUniformBuffers()
+{
+	title = "HPP Dynamic uniform buffers";
+}
+
+HPPDynamicUniformBuffers ::~HPPDynamicUniformBuffers()
+{
+	if (get_device().get_handle())
+	{
+		if (ubo_data_dynamic.model)
+		{
+			aligned_free(ubo_data_dynamic.model);
+		}
+
+		// Clean up used Vulkan resources
+		// Note : Inherited destructor cleans up resources stored in base class
+		get_device().get_handle().destroyPipeline(pipeline);
+
+		get_device().get_handle().destroyPipelineLayout(pipeline_layout);
+		get_device().get_handle().destroyDescriptorSetLayout(descriptor_set_layout);
+	}
+}
+
+// Wrapper functions for aligned memory allocation
+// There is currently no standard for this in C++ that works across all platforms and vendors, so we abstract this
+void *HPPDynamicUniformBuffers::aligned_alloc(size_t size, size_t alignment)
+{
+	void *data = nullptr;
+#if defined(_MSC_VER) || defined(__MINGW32__)
+	data = _aligned_malloc(size, alignment);
+#else
+	int res = posix_memalign(&data, alignment, size);
+	if (res != 0)
+		data = nullptr;
+#endif
+	return data;
+}
+
+void HPPDynamicUniformBuffers::aligned_free(void *data)
+{
+#if defined(_MSC_VER) || defined(__MINGW32__)
+	_aligned_free(data);
+#else
+	free(data);
+#endif
+}
```

Comment:
I know aligned_alloc/aligned_free is copied from the already existing sample.  However, it would be nice to do something a bit more cross platform.  Unfortunately Windows has problems with the STL aligned_alloc so it's probably better to use aligned_storage and aligned.
Fixing this isn't a requirement to do in this sample, but if you're willing, it would be nice.
in any case, both of these functions can be made static.

Reply:
I've made those two functions static.
But what do you mean with 

> it would be nice to do something a bit more cross platform.

How should that be "fixed"?

Reply:
an ideal fix in my opinion would be std::aligned_storage / std::aligned use such that there's no need for macro platform separated code.
Here's a quick search on SO:
https://stackoverflow.com/questions/32133203/what-can-i-use-instead-of-stdaligned-alloc-in-ms-visual-studio-2013

Reply:
I think, I'll leave that for others to handle.